sampleSpecs/test.smithy

```diff
@@ -194,3 +193,49 @@ structure ErrorDetails {
     @required
     location: String
 }
+
+
+@http(method: "PUT", uri: "/json-name-test")
+@httpRequestTests([
+    {
+        id: "JsonNameTest",
+        protocol: simpleRestJson,
+        documentation: "This is a test of the jsonName trait",
+        method: "PUT",
+        uri: "/json-name-test",
+        body:
+        """
+              {
+               "foo": "test"
+                }""",
+        bodyMediaType: "application/json",
+        params: {
+            bar: "test"
+        }
+    }
+])
+@httpResponseTests([
+    {
+        id: "JsonNameTest",
+        protocol: simpleRestJson,
+        documentation: "This is a test of the jsonName trait",
+        code: 200,
+        body:
+        """
+              {
+               "bar": "test"
+                }""",
+        bodyMediaType: "application/json",
+        params: {
+            foo: "test"
```

Comment:
Alright, so what's happening is that the DocumentDecoder/Encoder actually do support `@jsonName`, but these traits should NOT be taken into consideration when parsing the `params`. 

The first test doesn't pass because the `bar` parameter ends up being overlooked (because the DocumentDecoder is looking for `foo` when it shouldn't. Therefore the value associated for `bar` is set to None, which means that the body that gets received is an empty json, which doesn't match the expected body set by the test. 

The second test doesn't pass because it successfully decodes the input when it shouldn't, associating the `foo` param to the `bar` field of the `JsonNameTestInputOutput`, which ends up being `Some("test")`. Then, the jsoniter encoder does the right thing and encodes the `bar` field using the `foo` key, as per indicated by the `@jsonName("foo")`. 

So essentially, the fix would probably be to remove all hints from the schema before compiling the document decoder that is used to decode the params (but keep the timestamp-related hint that you already applied everywhere) .

Reply:
To summarise : the first test is correct but fails because of a bug in the implementation, the second test is incorrect

Reply:
Thanks for explaining

Reply:
> I think it's weird that DocumentDecoder looks at jsonName tbh

It would break least surprise if it didn't. Moreover Document is massively useful in a lot of usecases that involve existing json payloads that couldn't be expressed.